janusgraph-core/src/main/java/org/janusgraph/graphdb/management/ConfigurationManagementGraph.java

```diff
@@ -303,6 +308,9 @@ private void createIndexIfDoesNotExist(String indexName, String propertyKeyName,
             }
             management.commit();
             graph.tx().commit();
+        } else {
```

Comment:
So, `management.rollback()` and `graph.tx().rollback()` are now executed for both branches of the `if` statement and additionally at the end of the `catch` block. How about just moving these two lines to the end of the method? Then it could be removed from the three places it is currently in.

Reply:
Ok, I wrap potentially exception block in try/finllay

Reply:
Wait, why do you need to change the scope of the try block for that and why is `finally` needed? All I meant is that you can move the two lines out of the different branches to the end of the method.
You had it like this before:

```java
                ); // log.warn()
                management.rollback();
                graph.tx().rollback();
            } // catch
            management.commit();
            graph.tx().commit();
        } else {
            management.rollback();
            graph.tx().rollback();
        }
```

and my suggestion was to do it like this:

```java
        ); // log.warn()
    } // catch
} // if
management.rollback();
graph.tx().rollback();
```

This has exactly the same execution flow as in your previous version, just with less complexity.

With your recent change, you increased the scope of the `try` block a lot which shouldn't be necessary.

Reply:
We need commit block for initial case when property/index firstly created, but we can't commit if any statements failed or index already exists

Increase try/finally make method more robust, because any exceptions may occur in strange case ex. when property exists with same name but different types/cardinality and indexBuild failed

Reply:
Wow, I'm just realizing that I overlooked completely that in one case `rollback` is used here and `commit` in the other. So, my suggestion didn't make any sense at all here. Sorry about that!

> any exceptions may occur in strange case ex. when property exists with same name but different types/cardinality

But I'm not sure whether it's a good idea to just catch an exception like this and only log a warning. This sounds more like an error case that really has to be handled by the user.

So, I would suggest that you just change it back to what you had in your first version here. Sorry again for the inconvenience! I really should have given this a more detailed look.

Reply:
Ok, I'm revert